nimsuggest/nimsuggest.nim

```diff
@@ -581,6 +581,7 @@ proc processCmdLine*(pass: TCmdLinePass, cmd: string; conf: ConfigRef) =
         conf.projectName = a
       # if processArgument(pass, p, argsCount): break
 
+# TODO: refactor with nimconf.nim to avoid duplicating complex logic
```

Comment:
Good catch. But I cannot merge a PR that makes things worse, so then nimsuggest disagrees with nim's way of reading the configuration.

Reply:
ok, pending refactoring via OOP here https://github.com/nim-lang/Nim/pull/8682#issuecomment-416434086